dlc-manager/src/sub_channel_manager.rs

```diff
@@ -895,6 +1052,82 @@ where
         Ok(())
     }
 
+    /// Notify that LDK has decided to close the channel with given id.
+    pub fn notify_ln_channel_closed(&self, channel_id: ChannelId) -> Result<(), Error> {
+        let mut sub_channel = self
+            .dlc_channel_manager
+            .get_store()
+            .get_sub_channel(channel_id)?
+            .ok_or(Error::InvalidParameters(format!(
+                "No channel with id {:?} found",
+                channel_id
+            )))?;
+
+        let mut chain_monitor = self.dlc_channel_manager.get_chain_monitor().lock().unwrap();
+
+        let (updated_channel, updated_contract) = match sub_channel.state {
+            SubChannelState::Offered(_)
+            | SubChannelState::Accepted(_)
+            | SubChannelState::Confirmed(_)
+            | SubChannelState::CloseAccepted(_)
+            | SubChannelState::CloseConfirmed(_)
+            | SubChannelState::Finalized(_) => {
+                let dlc_channel_id = sub_channel
+                    .get_dlc_channel_id(0)
+                    .expect("to have a channel id");
+                let (closed_channel, closed_contract) =
+                    self.get_closed_dlc_channel_and_contract(dlc_channel_id, true)?;
+                sub_channel.state = SubChannelState::CounterOnChainClosed;
+                chain_monitor.cleanup_channel(sub_channel.channel_id);
+                chain_monitor.cleanup_channel(dlc_channel_id);
```

Comment:
It might be cleaner to do this in all cases after the match. We could `if let Some` the `dlc_channel_id`.

Reply:
Mmm I guess for now it'd be ok to do it, but ideally I think it should be differentiated because as I put in comments it's not always safe to cleanup right away.